src/chainstate/stacks/db/blocks.rs

```diff
@@ -3920,20 +3974,30 @@ impl StacksChainState {
                         panic!("Clarity VM believes it was running in 1.0: pre-Clarity.")
                     }
                     StacksEpochId::Epoch20 => {
+                        assert_eq!(
+                            sortition_epoch.epoch_id,
+                            StacksEpochId::Epoch2_05,
+                            "Should only transition from Epoch20 to Epoch2_05"
+                        );
```

Comment:
I think here we need to match on `sortition_epoch.epoch_id`, e.g.:

```rust
match sortition_epoch.epoch_id {
    StacksEpochId::Epoch2_05 => {
        receipts.push(clarity_tx.block.initialize_epoch_2_05()?);
        applied = true;
    }
    StacksEpochId::Epoch21 => {
        receipts.push(clarity_tx.block.initialize_epoch_2_05()?);
        receipts.push(clarity_tx.block.initialize_epoch_2_1()?);
        applied = true;
    }
    _ => panic!("Bad Stacks epoch transition; parent_epoch = {}, current_epoch = {}", ...),
}
```

This is because a miner could try mining off of a very deep ancestor block (i.e., a 2.0 block) from the 2.1 epoch, and would need to apply _both_ epoch transitions in that block.

Reply:
Seconded

Reply:
done